modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -458,6 +458,13 @@ def merge(self, right, **kwargs):
 
             kwargs["sort"] = False
 
+            # Want to ensure that these are python lists
+            if left_on is not None and right_on is not None:
```

Comment:
```suggestion
            if left_on and right_on:
```
fix also in other places

Reply:
it behaves differently depending on whether it's None or an empty list:
```python
>>> df1.merge(df2, on=[])
IndexError: list index out of range
>>> df1.merge(df2, on=None)
  col0    col1 col0
     a  b    a    c
0    1  2    3    2
1    2  3    4    3
2    3  4    5    4
3    4  5    6    5
```